content/guides/astro.md

```diff
@@ -17,11 +17,172 @@ comments: false
+
 If you need an example source code, get [Astrowind](https://github.com/onwidget/astrowind) (you'll need [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and [Node.js](https://nodejs.org/en/learn/getting-started/how-to-install-nodejs)):
 ```bash
 git clone https://github.com/onwidget/astrowind myStaticApp
 ```
 
+## Port and host
+
+Applications on Clever Cloud listen on port **8080**. If your project requires this configuration, set your port and host in Astro in one of two locations:
+
+{{< tabs items="package.json scripts,astro.config.mjs" >}}
+
+  {{< tab >}}
+
+   ```json {filename="package.json"}
+    "scripts": {
+      "dev": "astro dev",
+      "start": "astro dev",
+      "build": "astro check && astro build",
+      "preview": "astro preview --host 0.0.0.0 --port 8080",
+      "astro": "astro"
+    } 
+    ```
+  {{< /tab >}}
+
+  {{< tab >}}
+
+   ```javascript {filename="astro.config.mjs"}
+    import { defineConfig } from 'astro/config';
+
+    export default defineConfig({
+      server: {
+        port: 8080,
+        host: true
+      }
+    });
+    ```
+
+  {{< /tab >}}
+
+
+{{< /tabs >}}
+
+## Deploy Astro from the Console
+
+To deploy your Astro project to Clever Cloud, you will need to **create a new application**. The application wizard will walk you through the necessary configuration steps.
+
+{{% steps %}}
+
+### Create the application 
+
+From the lateral menubar, click **Create** > **An application**
+
+### Choose how to deploy:
+
+ - **Create a brand new app**: to deploy from a local repository with Git
+
+    or
+
+ - **Select a GitHub repository**: to deploy from GitHub
+
+### Select the type of application
+
+Select a **Node.js** application, or a **static** one.
+
+### Set up instance minimal size and scalability
+
+Set up the minimal size for your instance and scalability options. Astro sites can typically be deployed using the **Nano** instance. Depending on your project's specifications and dependencies, you may need to adjust accordingly as you watch the metrics from the **Overview** page.
+
+### Select the region
+
+Select a **region** to deploy your instance.
+
+### Connected services
+
+Skip [connecting **Add-ons** to your Clever application](/doc/addons/) unless you're using a database or Keycloak.
+
+### Inject **environment variables**
+
+For **Node.js**, no specific environment variable is needed to deploy Astro if you're using **npm**. If you're using **yarn** or **pnpm**, or deploying on a Static application, set the following environment variables.
+
+{{% /steps %}}
+
+#### Node.js variables
+
+{{< tabs items="pnpm,yarn" >}}
+  
+  {{< tab >}}
+
+  ```shell
+    CC_NODE_BUILD_TOOL="custom"
+    CC_PRE_BUILD_HOOK="npm install -g pnpm && pnpm install"
+    CC_CUSTOM_BUILD_TOOL="pnpm run astro telemetry disable && pnpm build"
+    CC_RUN_COMMAND="pnpm run preview"
```

Comment:
From [Astro documentation](https://docs.astro.build/en/reference/cli-reference/#astro-preview) regarding the `preview` command: 

> This command allows you to preview your site locally [after building](https://docs.astro.build/en/reference/cli-reference/#astro-build) to catch any errors in your build output before deploying it. It is not designed to be run in production.

Reply:
Exactly [what I submitted](https://github.com/withastro/docs/pull/9030#discussion_r1752270277) to @TheOtterlord but it seems that no command seems to be designed to run on production? I'm confused as well lol

Reply:
Astro is not made to serve the website directly. It's why it's often used to build them and then it's served with a dedicated static web server (Apache on Clever Cloud for now in the static runtime for example), or through adapters in SSR.

Reply:
the server from the node adapter can be run directly with `node ./dist/server/entry.mjs`